hub.py

```diff
@@ -234,9 +234,10 @@ def get_generated_config(self, auth_provider, proxy_secret_key):
         #
         # Allow explicilty ignoring auth0 setup
         if self.spec['auth0'].get('enabled', True):
+            auth0_org_domain = next(domain for domain in self.spec['domain'] if "2i2c" in domain)
             client = auth_provider.ensure_client(
                 self.spec['name'],
-                self.spec['domain'],
+                auth0_org_domain,
```

Comment:
I had forgotten that auth0 requires domain name the way we have set it up. I don't want to assume that there'll *always* be a 2i2c domain, and so this might be better explicit.

In this case, how about we:

1. If there's only one domain, it's listed as a string in `domain`.
2. If there's more than one domain, `domain` can be a list - but then you *must* specify `domain` under `auth0` as well to specify what domain should be used with auth0.

`auth0.domain` takes precedence over `domain` if it exists.

This way, we have minimal config for the general case, but *explicit* config for the more complex case.